Cargo.toml

```diff
@@ -18,9 +18,10 @@ http-cache-reqwest = { version = "0.12.0", features = [
 moka = { version = "0.12.3", default-features = false, features = [
     "future",
 ], optional = true }
-hyper-rustls = { version = "0.24.2", optional = true }
-rustls = { version = "0.21.9", optional = true }
+hyper-rustls = {version = "0.25.0", optional = true}
+rustls = {version = "0.22.2",optional = true}
 rustls-pemfile = { version = "1.0.4", optional = true }
+rustls-pki-types = "1.1.0"
```

Comment:
Why do we need this?

Reply:
Because since rustls 0.22.0 the types for Certificate and PrivateKey have been moved to this crate
```
use rustls_pki_types::{CertificateDer, PrivateKeyDer, PrivatePkcs1KeyDer, PrivatePkcs8KeyDer, PrivateSec1KeyDer};
```

Reply:
Ok. Do we need the new cert and keys though? Can we not reuse the old ones?

Reply:
The old ones have been removed from rustls crate so the only way is to add this crate. I followed the changes suggested in the **Breaking Change** section of this page https://github.com/rustls/rustls/releases/tag/v/0.22.0